erpnext/controllers/buying_controller.py

```diff
@@ -73,10 +73,14 @@ def set_supplier_from_item_default(self):
 
 	def validate_stock_or_nonstock_items(self):
 		if self.meta.get_field("taxes") and not self.get_stock_items():
-			tax_for_valuation = [d.account_head for d in self.get("taxes")
+			tax_for_valuation = [d for d in self.get("taxes")
 				if d.category in ["Valuation", "Valuation and Total"]]
+
 			if tax_for_valuation:
-				frappe.throw(_("Tax Category can not be 'Valuation' or 'Valuation and Total' as all items are non-stock items"))
+				for d in tax_for_valuation:
+					d.db_set("category", 'Total')
+				msgprint("Tax Category has been changed to \"Total\" because all Items in the Purchase Order "
```

Comment:
all messages have to be translatable _("Tax ...")

Also translatable messages cannot be broken into separate lines, and you cannot escape `\` inside translatable messages.